test/java/seedu/addressbook/data/person/NameTest.java

```diff
@@ -20,11 +20,12 @@ public void setup() throws IllegalValueException {
     }
     
     @Test
-    public void test_isSimilar_true() throws IllegalValueException {
+    public void test_isSimilar_true() {
         assertFalse(name.isSimilar(null));
         assertTrue(name.isSimilar(new Name("John K Smith")));
         assertTrue(name.isSimilar(new Name("John K SMITH")));
         assertTrue(name.isSimilar(new Name("John Smith")));
+        assertTrue(name.isSimilar(new Name("Smith, John K")));
```

Comment:
You should put all these test cases in different methods with clear method name. If you do it like this, later you might forgot what you want to test at each lines above.